dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/FontStyle.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2004-2020, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.common;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+import java.io.Serializable;
+
+import org.hisp.dhis.schema.annotation.PropertyRange;
+
+import static org.hisp.dhis.common.DxfNamespaces.DXF_2_0;
+
+/**
+ * Class representing various text styling properties.
+ *
+ * @author Lars Helge Overland
+ */
+@JacksonXmlRootElement( localName = "fontStyle", namespace = DXF_2_0 )
+public class FontStyle
+    implements Serializable
+{
+    private Font font;
+
+    private Integer fontSize;
+
+    private Boolean bold;
```

Comment:
Can we use `boolean` here? Or is it a Jackson requirement?

Reply:
The thinking is to use object properties consistently here to be able to represent the state of "no value". I.e. Integer (not int) for font size because it is hard to set a sensible default value/the default value might be client specific. Admittedly, we might want to have bold = false as a default, but again, it might be useful for the client to differentiate between an explicit "no bold" value and no value at all. This is debatable but I was landing on object properties.

Reply:
Ok, got it